File: c++/21_Merge_Two_Sorted_Lists.cpp
#include <iostream>
#include <vector>
#include <map>
#include <cmath>
#include <string>

using namespace std;

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
    	ListNode *head = new ListNode(0);
    	ListNode *tmp = head;
    	while (l1 && l2) {
    		if (l1->val < l2->val) {
    			tmp->next = l1;
    			l1 = l1->next;
    		}
    		else{
    			tmp->next = l2;
    			l2 = l2->next;
    		}
    		tmp = tmp->next;
    	}
    	// useful commands
    	tmp->next = l1 ? l2 : l1;
    	return head->next;
    }
};


int main() {
    Solution s;
    
}

File: python/21_Merge_Two_Sorted_Lists.py
#!/usr/bin/python
import sys
import os

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head = ListNode(0)
        tmp = head
        while l1 and l2:
        	if l1.val < l2.val:
        		tmp.next = l1
        		l1 = l1.next
        	else:
        		tmp.next = l2
        		l2 = l2.next
            tmp = tmp.next
        tmp.next = l1 if l1 else l2
        return head.next

def main():
    s = Solution()

if __name__ == "__main__":
    main()
